android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/WorkerBinder.kt

```diff
@@ -21,15 +21,164 @@ import com.uber.rib.core.lifecycle.InteractorEvent
 import com.uber.rib.core.lifecycle.PresenterEvent
 import com.uber.rib.core.lifecycle.WorkerEvent
 import io.reactivex.Observable
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.channels.BufferOverflow
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.merge
 import kotlinx.coroutines.flow.transformWhile
+import kotlinx.coroutines.launch
 import kotlinx.coroutines.rx2.asFlow
 import kotlinx.coroutines.rx2.asObservable
+import kotlin.system.measureTimeMillis
 
 /** Helper class to bind to an interactor's lifecycle to translate it to a [Worker] lifecycle. */
 object WorkerBinder {
+
+  /**
+   * A set of [RibWorkerBinderDurationListener]
+   */
+  @JvmStatic
+  val ribWorkerBinderDurationListeners: MutableSet<RibWorkerBinderDurationListener> = mutableSetOf()
+
+  /**
+   * Bind a worker (ie. a manager or any other class that needs an interactor's lifecycle) to an
+   * interactor's lifecycle events.
+   *
+   * Worker.onStart/onStop will run on the specified [ThreadingType]
+   *
+   * NOTE: If worker overrides ThreadingType other than ThreadingType.CALLER_THREAD. Worker threading
+   * type will take priority over the one specified at binding level
+   *
+   * @param worker The class that wants to be informed when to start and stop doing work.
+   * @param workerLifecycle Observable<WorkerEvent>
+   * @param threadingType ThreadingType. Threading type where worker will operate.
+   */
+  @JvmStatic
+  fun bind(
+    worker: Worker,
+    workerLifecycle: Observable<WorkerEvent>,
+    coroutineScope: CoroutineScope,
+    threadingType: ThreadingType = ThreadingType.BACKGROUND_COMPUTATION
+  ) {
+
+    var workerThreadingType = threadingType
+    if (worker.threadingType() != ThreadingType.CALLER_THREAD) {
+      workerThreadingType = worker.threadingType()
+    }
+
+    val coroutineDispatcher = mapToCoroutineDispatcher(workerThreadingType)
+    coroutineScope.launch(coroutineDispatcher) {
+      bindToWorkerLifecycle(workerLifecycle, worker, workerThreadingType)
+    }
+  }
+
+  /**
+   * Bind a worker (ie. a manager or any other class that needs an interactor's lifecycle) to an
+   * interactor's lifecycle events.
+   *
+   * Worker.onStart/onStop will run on the specified [ThreadingType]
+   *
+   * NOTE: If worker overrides ThreadingType other than ThreadingType.CALLER_THREAD. Worker threading
+   * type will take priority over the one specified at binding level
+   *
+   * @param worker The Worker to be bound
+   * @param interactor Interactor
+   * @param threadingType ThreadingType. Threading type where worker will operate.
+   */
+  @JvmStatic
+  fun bind(
+    worker: Worker,
+    interactor: Interactor<*, *>,
+    threadingType: ThreadingType
```

Comment:
Still debating if we should do this other than caller thread. Given the existing binding led to some misconceptions where Workers will be running upon. Maybe we make this new binders be defaulted to background